### PR TITLE
Improve object handler for CDS metadata extension (DDLX) 

### PR DESCRIPTION
**Pull request by ThomasPloski**
The existing object handler for CDS metadata extension (DDLX) has been reviewed by an expert.
Doing this, the following feedback has been provided:

enable existing logging capability
include environment specific logic (e.g. import of an inactive version)
improve exception handling so that the related object information can be displayed in log
assign package information to handle abap language version ina proper way
adapt lock object

### CODE REVIEW

File: src/objects/zcl_abapgit_object_ddlx.clas.abap [hunk 1]
```diff
@@ -279,12 +294,33 @@ CLASS ZCL_ABAPGIT_OBJECT_DDLX IMPLEMENTATION.
...
 
         li_persistence = get_persistence( ).
-        li_persistence->get(
-          EXPORTING
-            p_object_key  = lv_object_key
-            p_version     = swbm_version_active
-          CHANGING
-            p_object_data = li_data_model ).
+
+        IF zcl_abapgit_factory=>get_environment( )->compare_with_inactive( ) = abap_true.
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_ddlx.clas.abap]: will not compile until https://github.com/larshp/abapGit/pull/2809 is merged

File: src/objects/zcl_abapgit_object_ddlx.clas.xml [hunk 2]
```diff
@@ -11,6 +11,14 @@
     <FIXPT>X</FIXPT>
     <UNICODE>X</UNICODE>
    </VSEOCLASS>
+   <DESCRIPTIONS>
+    <SEOCOMPOTX>
+     <CLSNAME>ZCL_ABAPGIT_OBJECT_DDLX</CLSNAME>
+     <CMPNAME>MO_TADIR_INTERFACE</CMPNAME>
+     <LANGU>E</LANGU>
+     <DESCRIPT>Boolean Variable (X = True, - = False, Space = Unknown)</DESCRIPT>
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_ddlx.clas.xml]: please remove description

File: src/objects/zcl_abapgit_objects_super.clas.abap [hunk 3]
```diff
@@ -308,11 +308,12 @@ CLASS ZCL_ABAPGIT_OBJECTS_SUPER IMPLEMENTATION.
     ls_e071_tadir-obj_name = ms_item-obj_name.
     INSERT ls_e071_tadir INTO TABLE lt_e071_tadirs.
 
-    CALL FUNCTION 'RS_INACTIVE_OBJECTS_WARNING'
+CALL FUNCTION 'RS_INACTIVE_OBJECTS_WARNING'
```
Comment (larshp) [inline src/objects/zcl_abapgit_objects_super.clas.abap]: run pretty printer, indentation wrong, also `phased_activation` is a optional parameter, so no need to add it

File: src/objects/zif_abapgit_object.intf.abap [hunk 4]
```diff
@@ -2,34 +2,36 @@ INTERFACE zif_abapgit_object
...
-      !iv_package TYPE devclass
-      !io_xml     TYPE REF TO zcl_abapgit_xml_input
-      !iv_step    TYPE ty_deserialization_step
-      !ii_log     TYPE REF TO zif_abapgit_log
+      !iv_package               TYPE devclass
+      !io_xml                   TYPE REF TO zcl_abapgit_xml_input
+      !iv_step                  TYPE ty_deserialization_step
+      !ii_log                   TYPE REF TO zif_abapgit_log
+    RETURNING
+      VALUE(rv_complete_status) TYPE ty_complete_status
```
Comment (larshp) [inline src/objects/zif_abapgit_object.intf.abap]: this is a central change to overall logic, please make sure to include it in the description of the pull request

File: src/zcl_abapgit_objects.clas.abap [hunk 5]
```diff
@@ -1083,8 +1084,12 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
     li_obj = create_object( is_item     = rs_files_and_item-item
                             iv_language = iv_language ).
     li_obj->mo_files = lo_files.
+
     CREATE OBJECT lo_xml.
-    li_obj->serialize( lo_xml ).
+    CREATE OBJECT lo_log.
+
+    li_obj->serialize( io_xml      = lo_xml
+                       ii_log      = lo_log ). "to be replaced by item logger
```
Comment (larshp) [inline src/zcl_abapgit_objects.clas.abap]: also a change to central logic, `lo_log` will be garbage collected, so any potential errors will not be shown to the user?

File: src/zcl_abapgit_serialize.clas.abap [hunk 6]
```diff
@@ -83,7 +83,7 @@ CLASS ZCL_ABAPGIT_SERIALIZE IMPLEMENTATION.
 
     lo_settings = zcl_abapgit_persist_settings=>get_instance( )->read( ).
 
-    IF zcl_abapgit_environment=>is_merged( ) = abap_true
+    IF zcl_abapgit_factory=>get_environment( )->is_merged( ) = abap_true
```
Comment (larshp) [inline src/zcl_abapgit_serialize.clas.abap]: does not compile until https://github.com/larshp/abapGit/pull/2809 is merged

File: src/objects/zcl_abapgit_object_ddlx.clas.abap (hunk 1 above)
Comment (ThomasPloski) [reply src/objects/zcl_abapgit_object_ddlx.clas.abap]: What's to do? Wait till #2809 is merged?

File: src/objects/zcl_abapgit_object_ddlx.clas.xml (hunk 2 above)
Comment (ThomasPloski) [reply src/objects/zcl_abapgit_object_ddlx.clas.xml]: done

File: src/objects/zif_abapgit_object.intf.abap (hunk 4 above)
Comment (ThomasPloski) [reply src/objects/zif_abapgit_object.intf.abap]: done

File: src/objects/zcl_abapgit_object_ddlx.clas.abap (hunk 1 above)
Comment (larshp) [reply src/objects/zcl_abapgit_object_ddlx.clas.abap]: yes

Comment (larshp) [reply src/objects/zcl_abapgit_object_ddlx.clas.abap]: compare_with_inactive is also a new method that is not part of #2809 ?

File: src/objects/zcl_abapgit_object_ddlx.clas.abap [hunk 7]
```diff
@@ -110,6 +110,16 @@ CLASS ZCL_ABAPGIT_OBJECT_DDLX IMPLEMENTATION.
...
 
 
+  METHOD get_tadir_interface.
+    IF mo_tadir_interface IS NOT BOUND.
+      CREATE OBJECT mo_tadir_interface TYPE lcl_tadir
+        EXPORTING
+          io_outer = me.
+    ENDIF.
+    ri_tadir = mo_tadir_interface.
+  ENDMETHOD.
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_ddlx.clas.abap]: this method `get_tadir_interface` seems like a workaround? suggest leaving this out and opening a new issue regarding dependency injection for `tadir_insert`

File: src/objects/zcl_abapgit_objects_super.clas.abap (hunk 3 above)
Comment (ThomasPloski) [reply src/objects/zcl_abapgit_objects_super.clas.abap]: done

**Comment by ThomasPloski** [comment]
a new one needs to be created due to different changes

**Comment by ThomasPloski** [comment]
reopen with req. changes

File: src/objects/zcl_abapgit_object_ddlx.clas.abap (hunk 7 above)
Comment (ThomasPloski) [reply src/objects/zcl_abapgit_object_ddlx.clas.abap]: done

File: src/objects/zif_abapgit_object.intf.abap (hunk 4 above)
Comment (larshp) [inline src/objects/zif_abapgit_object.intf.abap]: this change is unrelated to DDLX changes? `rv_complete_status`

File: src/zcl_abapgit_objects.clas.abap [hunk 8]
```diff
@@ -1113,8 +1114,12 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
...
```
Comment (larshp) [inline src/zcl_abapgit_objects.clas.abap]: separate not related change

File: src/objects/zcl_abapgit_objects_super.clas.abap [hunk 9]
```diff
@@ -313,6 +313,7 @@ CLASS ZCL_ABAPGIT_OBJECTS_SUPER IMPLEMENTATION.
         suppress_protocol         = abap_false
         with_program_includes     = abap_false
         suppress_dictionary_check = abap_false
+        phased_activation         = abap_false
```
Comment (larshp) [inline src/objects/zcl_abapgit_objects_super.clas.abap]: suggest keeping out this change, but this should not change anything, it just adds the parameter default to the code
